test/testlib/nuodb_database_utilities.go

```diff
@@ -55,6 +56,14 @@ func GetExtractedOptions(options *helm.Options) (opt ExtractedOptions) {
 	return
 }
 
+func EnsureDatabaseNotRunning(t *testing.T, adminPod string, opt ExtractedOptions, kubectlOptions *k8s.KubectlOptions) {
+	// invoke shutdown database; this may fail if the database is already NOT_RUNNING, which is okay
+	k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", adminPod, "--", "nuocmd", "shutdown", "database", "--db-name", opt.DbName)
+	// wait for all database processes to exit
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", adminPod, "--", "nuocmd", "check", "database", "--db-name", opt.DbName, "--num-processes", "0", "--timeout", "30")
+	assert.NilError(t, err, output)
```

Comment:
use `k8s.RunKubectl` for both. You do not need the output, since you are not parsing it. You also do not need the error message (`RunKubectlE`), since you are calling NilError on it, without additional parsing.

`RunKubectl` is this under the hood:
`require.NoError(t, RunKubectlE(t, options, args...))`

Reply:
It is good to have the output of `nuocmd check database` so that the assert generates a useful message. I'll update the `nuocmd shutdown database` invocation to use `k8s.RunKubectlE()`.

Reply:
you can see the output in the logs regardless of which function you use

Reply:
Before I added the output to the assert I had no clue why the test was failing. Even with the knowledge that the command logging is somewhere else in this massive ream of log messages, I still have to look for it. It saves human time to just have the message in the assert. Is there any reason not to do that in `k8s.RunKubectl()`?

Reply:
Thats a good argument for using `k8s.RunKubectlE()`. I buy it.

Again, we do not own the code for `k8s.RunKubectl()`

Reply:
Ah, okay. I'll just make the change you suggested and then we can think about adding our own helper functions to generate better assert messages.